File: subnetting.py
#import pandas as pd
def print_table(data):
    widths = {k: max(len(k), *(len(str(d[k])) for d in data)) for k in data[0]}
    print(" ".join(f"{k:{widths[k]}}" for k in widths))
    print("-" * sum(widths.values()))
    for row in data:
        print(" ".join(f"{str(row[k]):{widths[k]}}" for k in widths))

MAGIC = 8
mgk_net = [["/" + str(MAGIC * (j - 1) + i) for j in range(1, int(MAGIC / 2) + 1)] for i in range(1, MAGIC + 1)]
nets, adrs = zip(*[(2 ** (i + 1), 2 ** (MAGIC - i - 1)) for i in range(MAGIC)])
networks = []
for i in range(MAGIC):
    n = {}
    n['index'] = i + 1
    n['cidr'] = mgk_net[i]
    n['last_subnet'] = MAGIC * int(MAGIC * MAGIC / 2) - adrs[i]
    n['nets'] = nets[i]
    n['addresses'] = adrs[i]
    n['range'] = f'[1:{adrs[i] - 2}]'
    n['gateway'] = adrs[i] - 1
    n['next_network'] = [x for x in range(0, 256, adrs[i])][1:11]
    n['network_calc'] = f'{adrs[i]}*(n<={nets[i] - 1})'
    networks.append(n)
#df_networks = pd.DataFrame(networks)
#print(df_networks)
print_table(networks)

File: subnetting.js
function subnetting(MAGIC=8) {
   const mgk_net = Array.from({ length: MAGIC }, (_, i) => Array.from({ length: Math.floor(MAGIC / 2) }, (_, j) => `/${(j + 1) * MAGIC - MAGIC + i + 1}`));
   const [nets, adrs] = Array.from({ length: MAGIC }, (_, i) => [2 ** (i + 1), 2 ** (MAGIC - i - 1)]).reduce(([nets, adrs], [net, adr]) => [[...nets, net], [...adrs, adr]], [[], []]);

   const networks = Array.from({ length: MAGIC }, (_, i) => ({
      index: i + 1,
      cidr: mgk_net[i],
      last_subnet: MAGIC * Math.floor(MAGIC * MAGIC / 2) - adrs[i],
      nets: nets[i],
      addresses: adrs[i],
      range: `[1:${adrs[i] - 2}]`,
      gateway: adrs[i] - 1,
      next_network: Array.from({ length: Math.floor((MAGIC * Math.floor(MAGIC * MAGIC / 2)-1) / adrs[i]) }, (_, j) => (j + 1) * adrs[i]).slice(0, 10),
      network_calc: `${adrs[i]}*(n<=${nets[i] - 1})`,
   }));
   console.table(networks);
}; subnetting();
